core/app/models/spree/promotion_code.rb

```diff
@@ -37,6 +38,10 @@ def usage_limit
     promotion.per_code_usage_limit
   end
 
+  def promotion_not_apply_automatically
+    errors.add(:base, I18n.t('spree.errors.messages.could_not_create_promotion_code_on_apply_automatically_promotion')) if promotion.apply_automatically
```

Comment:
There's no need to use a generic key here, you can have the same result by using:

```ruby
errors.add(:base, :disallaowed_with_apply_automatically) if promotion.apply_automatically
```
and moving the translation into:

```
en.activerecord.errors.models.spree/promotion_code.attributes.base.disallaowed_with_apply_automatically
```

(this probably needs a double check on documentation 🙂)

Reply:
Do you mean the solidus developers guide? In that guide there isn't any section about the promotion code, maybe I can create a new page  for the promotion codes.

Reply:
> this probably needs a double check on documentation

No, I meant in the Rails documentation.

Reply:
Sorry, my fault, I misunderstood 🙂